File: 2970. Count the Number of Incremovable Subarrays I.c
int incremovableSubarrayCount(int* nums, int numsSize) {
    int a=1;//开头严格增的序列长度  the length of the strictly increasing part at the head
    int b=1;//结尾严格增的序列长度  the length of the strictly increasing part at the end
    while(a<numsSize && nums[a-1]<nums[a]){
        a++;
    }
    while(b<numsSize && nums[numsSize-b]>nums[numsSize-b-1]){
        b++;
    }
    if(a==numsSize){//原序列严格增的情况  the original array is strictly increasing
        return a*(a+1)/2;
    }
    int ans=0;
    for(int i=0;i<a;i++){
        for (int j=0;j<b;j++){
            if (nums[i]<nums[numsSize-j-1]){
                ans++;//包含头尾的情况  the case that contains both the head and the end
            }
        }
    }
    ans+=a+b+1;//包含头或尾的情况以及空序列的情况  the case the contains only the head or the end or the empty array
    return ans;
}
//Runtime 0ms
//Memory 8.9MB

File: 2970. Count the Number of Incremovable Subarrays I.py
class Solution:
    def incremovableSubarrayCount(self, nums: List[int]) -> int:
        n=len(nums)
        head=[nums[0]]#记录开头的严格增部分  record the strictly increasing part in the head
        end=[nums[-1]]#记录结尾的严格增部分  record the strictly increasing part in the end
        i=1
        while i<n and nums[i]>nums[i-1]:
            head.append(nums[i])
            i+=1
        i=-2
        while i>=-n and nums[i]<nums[i+1]:
            end=[nums[i]]+end
            i-=1
        a=len(head)
        b=len(end)
        if a==n:#原序列严格增  the original list in strictly increasing
            return int(n*(n+1)/2)
        ans=0
        for i in range(a):#去除子数列后包含头尾的情况  the case that contains both head and end
            for j in range(b):
                if head[i]<end[-j-1]:
                    ans+=1
        ans+=a+b+1#只包含头尾和空数列  the case that contains only head or end or an empty list
        return ans
#Run time 0ms
#Memory 17.8MB
